test/example/SnapshotGateTest.js

```diff
@@ -0,0 +1,717 @@
+        await foreign20.connect(holder).mint(holder.address, amountToMint);
+        await foreign20.connect(holder).approve(snapshotGate.address, amountToMint);
+
+        // Create snapshot entry for holder / token
+        for (let i = 1; i <= snapshotTokenCount; i++) {
+          // The token id
+          const tokenId = i.toString();
+
+          // Get a random balance 1 - 9
+          const balance = Math.floor(Math.random() * 10) + 1;
+
+          // Track the total supply of each token - corresponding offer's qty available must match
+          snapshotTokenSupplies[tokenId] = String(Number(snapshotTokenSupplies[tokenId] || 0) + balance);
+
+          // Add snapshot entry
+          snapshot.push({
+            owner: holder.address,
+            tokenId: i.toString(),
+            amount: balance.toString(),
+          });
+        }
+      }
+
+      // Create gated offers in a loop
+
+      offers = [];
+      groups = [];
+
+      // Make 2 passes, creating native token offers and then ERC20 offers
+      for (let j = 0; j < 2; j++) {
+        for (let i = 1; i <= snapshotTokenCount; i++) {
+          // The token id
+          const tokenId = i.toString(); // first and second batches use same token ids
+          offerId = Number(snapshotTokenCount * j + i).toString(); // offer id from first or second batch
+          groupId = offerId;
+
+          // The supply of this token
+          const tokenSupply = snapshotTokenSupplies[tokenId];
+
+          // Create the offer
+          const mo = await mockOffer();
+          ({ offerDates, offerDurations } = mo);
+          offer = mo.offer;
+          price = offer.price;
+
+          // Set price in ERC-20 token if on second pass
+          if (j > 0) {
+            offer.exchangeToken = foreign20.address;
+            offer.buyerCancelPenalty = "0";
+          }
+
+          offer.sellerDeposit = "0";
+          offer.quantityAvailable = tokenSupply;
+          disputeResolverId = disputeResolver.id;
+
+          // Check if entities are valid
+          expect(offer.isValid()).is.true;
+          expect(offerDates.isValid()).is.true;
+          expect(offerDurations.isValid()).is.true;
+
+          // Create the offer
+          await offerHandler
+            .connect(operator)
+            .createOffer(offer, offerDates, offerDurations, disputeResolverId, agentId);
+          offers.push(offer);
+
+          // Required constructor params for Group
+          offerIds = [offerId];
+
+          // Create Condition
+          condition = mockCondition({
+            tokenAddress: snapshotGate.address,
+            threshold: "0",
+            maxCommits: tokenSupply,
+            tokenType: TokenType.NonFungibleToken,
+            tokenId: tokenId,
+            method: EvaluationMethod.SpecificToken,
+          });
+          expect(condition.isValid()).to.be.true;
+
+          // Create Group
+          group = new Group(groupId, seller.id, offerIds);
```

Comment:
It looks like 1 group per offer is being created instead of 1 group with all offer Ids. Is that what the scenario will be? Maybe it doesn't matter. I was just thinking we should test the actual scenario if possible.

Reply:
Each offer has a different token id that is supposed to gate it. This is the actual scenario.